src/primitives/transaction.cpp

```diff
@@ -9,6 +9,8 @@
 #include <tinyformat.h>
 #include <util/strencodings.h>
 
+#include <assert.h>
```

Comment:
Nit: `#include <assert.h>` is deprecated in C++. Use the idiomatic form `#include <cassert>` :)